veniq/baselines/semi/recommend.py

```diff
@@ -99,19 +102,27 @@ def _convert_ExtractionOpportunity_to_EMO(
     return (start_line_opportunity, start_line_opportunity + addit_lines_brackets)
 
 
-def check_format_validity(method_decl: List[str]) -> None:
+def check_input_method_format(method_decl: str) -> None:
     """
     Checks that the input method declaration is a syntactically
     correct Java method declaration.
     Raises exception otherwise.
     """
-    # TODO
-    pass
+    # check type
+    if type(method_decl) != str:
+        raise WrongInputToApi('Input should be a string with line breaks.')
+
+    # check it is a method declaration
+    class_interf_pattern = r"\s*(|(public|protected|private|static|)\s+)(class|interface)\s+[\w\W]+\s*"
+    class_interf_pattern_comp = re.compile(class_interf_pattern)
+    if class_interf_pattern_comp.match(method_decl):
+        raise WrongInputToApi('Input to recommend API should a method declaration, \
```

Comment:
Again, you mention `API`. Let's consider placing this logic in a web service

Reply:
ok